addOns/accessControl/src/main/java/org/zaproxy/zap/extension/accessControl/AccessControlAPI.java

```diff
@@ -0,0 +1,217 @@
+
+    private static final String PARAM_CONTEXT_ID = "contextId";
+    private static final String PARAM_USER_ID = "userId";
+    private static final String PARAM_RAISE_ALERT = "raiseAlert";
+    private static final String PARAM_ALERT_RISK_LEVEL = "alertRiskLevel";
+    private static final String PARAM_UNAUTH_USER = "scanAsUnAuthUser";
+    private static final String PARAM_FILENAME = "fileName";
+
+    private static final Logger LOGGER = Logger.getLogger(AccessControlAPI.class);
+
+    public AccessControlAPI(ExtensionAccessControl extension) {
+        this.extension = extension;
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_SCAN,
+                        new String[] {PARAM_CONTEXT_ID, PARAM_USER_ID},
+                        new String[] {
+                            PARAM_UNAUTH_USER, PARAM_RAISE_ALERT, PARAM_ALERT_RISK_LEVEL
+                        }));
+
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_WRITE_HTML_REPORT, new String[] {PARAM_CONTEXT_ID, PARAM_FILENAME}));
+
+        this.addApiView(new ApiView(VIEW_GET_SCAN_PROGRESS, new String[] {PARAM_CONTEXT_ID}));
+        this.addApiView(new ApiView(VIEW_GET_SCAN_STATUS, new String[] {PARAM_CONTEXT_ID}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        ApiResponse result = null;
+
+        switch (name) {
+            case ACTION_SCAN:
+                LOGGER.debug("Access control start scan called");
+
+                AccessControlScanStartOptions startOptions = new AccessControlScanStartOptions();
+
+                startOptions.targetContext = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID);
+
+                if (usersExtension == null) {
+                    usersExtension =
+                            Control.getSingleton()
+                                    .getExtensionLoader()
+                                    .getExtension(ExtensionUserManagement.class);
+                }
+
+                List<User> users = new ArrayList<User>();
+
+                String[] commaSeparatedUserIDs =
+                        ApiUtils.getNonEmptyStringParam(params, PARAM_USER_ID).split("\\s*,\\s*");
+
+                for (int i = 0; i < commaSeparatedUserIDs.length; i++) {
+                    int userID = Integer.parseInt(commaSeparatedUserIDs[i]);
+                    users.add(
+                            usersExtension
+                                    .getContextUserAuthManager(
+                                            startOptions.targetContext.getIndex())
+                                    .getUserById(userID));
+                }
+
+                startOptions.targetUsers = users;
+
+                // Add unauthenticated user
+                if (params.has(PARAM_UNAUTH_USER)
```

Comment:
Can use `params.optBoolean(PARAM_UNAUTH_USER, false)`.

Reply:
This should probably be obvious if I fully understood params.optBoolean, but did you mean just for the first condition or for all 3?

Reply:
It was for all 3.